src/repository.c

```diff
@@ -1341,28 +1347,47 @@ bool git_repository__reserved_names(
 }
 #endif
 
-static int check_repositoryformatversion(git_config *config)
+static int check_repositoryformatversion(git_config *config, int *version)
 {
-	int version, error;
+	int error;
 
-	error = git_config_get_int32(&version, config, "core.repositoryformatversion");
+	error = git_config_get_int32(version, config, "core.repositoryformatversion");
 	/* git ignores this if the config variable isn't there */
 	if (error == GIT_ENOTFOUND)
 		return 0;
 
 	if (error < 0)
 		return -1;
 
-	if (GIT_REPO_VERSION < version) {
+	if (GIT_REPO_MAX_VERSION < *version) {
 		git_error_set(GIT_ERROR_REPOSITORY,
 			"unsupported repository version %d. Only versions up to %d are supported.",
```

Comment:
I know you didn't create this problem, but while you're here, would you mind fixing this?

Let's not split sentences, since it can make application code that tries to display error messages weird.  They're often trying to incorporate our messages into their own and they're probably doing that by adding a period to the end themselves.  eg:

```
unsupported repository version %d; only versions up to %d are supported
```

Reply:
Sure, I can fix that.